File: 1580-shuffle-the-array/shuffle-the-array.java
class Solution {
    public int[] shuffle(int[] nums, int n) {

        int start = 0;
        int mid = n;

        int[] arr = new int[nums.length];

        for(int i = 0; i <= nums.length-1; i++) {
            if (i%2 == 0) {
                arr[i] = nums[start];
                start ++;
            }
            else {
                arr[i] = nums[mid];
                mid ++;  
            }
        }
        return arr;
    }
}

File: 1580-shuffle-the-array/shuffle-the-array.py
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        start = 0
        mid = n

        arr = [0] * (len(nums))

        for i in range(0, len( nums)):
            if i % 2 == 0:
                arr[i] = nums[start]
                start += 1
            else:
                arr[i] = nums[mid]
                mid += 1

        return arr